packages/internal/src/build/api.ts

```diff
@@ -1,85 +1,91 @@
+  // No need to wait for promise to resolve
+  BUILD_CTX?.dispose()
+  BUILD_CTX = null
+
+  return transpileApi(findApiFiles())
+}
 
-  const srcFiles = findApiFiles()
+export const rebuildApi = async () => {
+  const apiFiles = findApiFiles()
 
-  const prebuiltFiles = prebuildApiFiles(srcFiles).filter(
-    (path): path is string => path !== undefined
-  )
+  if (!BUILD_CTX) {
+    BUILD_CTX = await context(getEsbuildOptions(apiFiles))
+  }
 
-  return transpileApi(prebuiltFiles)
+  return BUILD_CTX.rebuild()
 }
 
-export const cleanApiBuild = () => {
+export const cleanApiBuild = async () => {
   const rwjsPaths = getPaths()
-  removeSync(rwjsPaths.api.dist)
-  removeSync(path.join(rwjsPaths.generated.prebuild, 'api'))
+  return remove(rwjsPaths.api.dist)
 }
 
-/**
- * Remove RedwoodJS "magic" from a user's code leaving JavaScript behind.
- */
-export const prebuildApiFiles = (srcFiles: string[]) => {
-  const rwjsPaths = getPaths()
-  const rwjsConfig = getConfig()
-  const plugins = getApiSideBabelPlugins({
-    openTelemetry:
-      rwjsConfig.experimental.opentelemetry.enabled &&
-      rwjsConfig.experimental.opentelemetry.wrapApi,
-  })
-
-  return srcFiles.map((srcPath) => {
-    const relativePathFromSrc = path.relative(rwjsPaths.base, srcPath)
-    const dstPath = path
-      .join(rwjsPaths.generated.prebuild, relativePathFromSrc)
-      .replace(/\.(ts)$/, '.js')
-
-    const result = prebuildApiFile(srcPath, dstPath, plugins)
-    if (!result?.code) {
-      // TODO: Figure out a better way to return these programatically.
-      console.warn('Error:', srcPath, 'could not prebuilt.')
-
-      return undefined
-    }
-
-    fs.mkdirSync(path.dirname(dstPath), { recursive: true })
-    fs.writeFileSync(dstPath, result.code)
-
-    return dstPath
-  })
+const runRwBabelTransformsPlugin = {
+  name: 'rw-esbuild-babel-transform',
+  setup(build: PluginBuild) {
+    const rwjsConfig = getConfig()
+
+    build.onLoad({ filter: /\.(js|ts|tsx|jsx)$/ }, async (args) => {
+      // @TODO Implement LRU cache? Unsure how much of a performance benefit its going to be
+      // Generate a CRC of file contents, then save it to LRU cache with a limit
+      // without LRU cache, the memory usage can be come unbound
+      const transformedCode = await transformWithBabel(
+        args.path,
+        getApiSideBabelPlugins({
+          openTelemetry:
+            rwjsConfig.experimental.opentelemetry.enabled &&
+            rwjsConfig.experimental.opentelemetry.wrapApi,
+        })
+      )
+
+      if (transformedCode?.code) {
+        return {
+          contents: transformedCode.code,
+          loader: 'js',
+        }
+      }
+
+      throw new Error(`Could not transform file: ${args.path}`)
+    })
+  },
+}
+
+export const transpileApi = async (files: string[]) => {
+  return build(getEsbuildOptions(files))
 }
 
-export const transpileApi = (files: string[], options = {}) => {
+function getEsbuildOptions(files: string[]) {
```

Comment:
```suggestion
function getEsbuildOptions(files: string[]): BuildOptions {
```

`BuildOptions` is imported from 'esbuild'